File: applications/luci-app-statistics/luasrc/statistics/rrdtool/definitions/ping.lua
-- Copyright 2008 Freifunk Leipzig / Jo-Philipp Wich <jow@openwrt.org>
-- Licensed to the public under the Apache License 2.0.

module("luci.statistics.rrdtool.definitions.ping", package.seeall)

function rrdargs( graph, plugin, plugin_instance, dtype )
	return {
		-- Ping roundtrip time
		{ title = "%H: ICMP Round Trip Time", vlabel = "ms",
		  number_format = "%5.1lf ms", data = {
			sources = { ping = { "value" } },
			options = { ping__value = { noarea = true, title = "%di" } }
		} },

		-- Ping droprate
		{ title = "%H: ICMP Drop Rate", vlabel = "%",
		  number_format = "%5.2lf %%", data = {
			types   = { "ping_droprate" },
			options = { ping_droprate = { title = "%di" } }
		} },

		-- Ping standard deviation
		{ title = "%H: ICMP Standard Deviation", vlabel = "ms",
		  number_format = "%5.2lf ms", data = {
			types   = { "ping_stddev" },
			options = { ping_stddev = { title = "%di" } }
		} },
	}
end

File: applications/luci-app-statistics/htdocs/luci-static/resources/statistics/rrdtool/definitions/ping.js
/* Licensed to the public under the Apache License 2.0. */

'use strict';
'require baseclass';

return baseclass.extend({
	title: _('Ping'),

	rrdargs: function(graph, host, plugin, plugin_instance, dtype) {
		var ping = {
			title: "%H: ICMP Round Trip Time",
			vlabel: "ms",
			number_format: "%5.1lf ms",
			data: {
				sources: {
					ping: [ "value" ]
				},
				options: {
					ping__value: {
						noarea: true,
						overlay: true,
						title: "%di"
					}
				}
			}
		};

		var droprate = {
			title: "%H: ICMP Drop Rate",
			vlabel: "%",
			number_format: "%5.2lf %%",
			data: {
				types: [ "ping_droprate" ],
				options: {
					ping_droprate: {
						noarea: true,
						overlay: true,
						title: "%di",
						transform_rpn: "100,*"
					}
				}
			}
		};

		var stddev = {
			title: "%H: ICMP Standard Deviation",
			vlabel: "ms",
			number_format: "%5.1lf ms",
			data: {
				types: [ "ping_stddev" ],
				options: {
					ping_stddev: {
						noarea: true,
						overlay: true,
						title: "%di"
					}
				}
			}
		};

		return [ ping, droprate, stddev ];
	}
});
